cips/cip-price-enforcement.md

```diff
@@ -0,0 +1,59 @@
+
+This proposal should be viewed as a sub component of a more broader effort and thus it’s scope is strictly focused towards the more foundational aspect of fee enforcement: **ensuring that the value captured goes to those that provided that value**. It does not pertain to actual pricing mechanisms, tipping, refunds, futures and other possible future works.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+All transactions MUST be decodable by the network. They MUST have a recognised signer, a signature that authenticates the signer, a fee and a gas. This does not imply that they need be completely valid, they just need a minimum degree of validity that allows the network to charge the signer for the allocation of gas for performing the computation, networking and storage.
+
+We define the gas price as the fee divided by the gas. In other words, this is the amount of utia paid per unit of gas. All transactions MUST have a gas price that is greater than or equal to a minimum network-wide agreed upon gas price.
+
+Both of these rules are block validity rules. Correct validators will vote `nil` or against the proposal if the proposed block contains any transaction that violates these two rules.
+
+Note that validators may in addition set their own constraints as to what they deem acceptable in a proposed block. For example, they may only propose blocks with a gas-price that is higher than the global minimum. However, correct validators, SHOULD gossip all valid transactions that are above the defined minimum gas price and not their own local price.
+
+This minimum gas price SHOULD be queryable by client implementations.
+
+## Rationale
+
+The primary rationale for this decision is to prevent the payment system for data availability from migrating off-chain and manifesting in secondary markets. As a concrete example, currently validators would earn more revenue if they convinced users to pay them out of band and set the transaction fee to 0 such that all revenue went to the proposer and none to the rest of the validators/delegators.
+
+Having the means to enforce payment also supports the introduction of more advanced payment algorithms in the future.
+
+Lastly, this change removes the possible incongruity that would form when it comes to gossiping transactions when consensus nodes use different minimum gas prices.
+
+## Backwards Compatibility
+
+This requires a modification to the block validity rules and is thus breaks the state machine. It will need to be introduced in a major release.
```

Comment:
```suggestion
This requires a modification to the block validity rules and thus breaks the state machine. It will need to be introduced in a major release.
```